src/openrct2-android/app/src/main/java/io/openrct2/GameActivity.java

```diff
@@ -1,15 +1,80 @@
 package io.openrct2;
 
+import android.icu.util.Currency;
+import android.icu.util.LocaleData;
+import android.icu.util.ULocale;
+import android.os.Build;
 import android.view.View;
 
 import org.libsdl.app.SDLActivity;
 
+import java.util.Locale;
+
 public class GameActivity extends SDLActivity {
 
     public float getDefaultScale() {
         return getResources().getDisplayMetrics().density;
     }
 
+    public String getDefaultLocale(String[] supportedTags) {
+        Locale deviceLocale;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            deviceLocale = getResources().getConfiguration().getLocales().get(0);
+        } else {
+            deviceLocale = getResources().getConfiguration().locale;
+        }
+
+        for (String supportedTag : supportedTags) {
+            if (supportedTag.isEmpty()) continue;
+            String[] splits = supportedTag.split("-");
+            String language = splits[0];
+            String country = splits[1];
+            if (deviceLocale.getLanguage().equals(language) && deviceLocale.getCountry().equals(country)) {
+                return supportedTag;
+            }
+        }
+
+        Locale canadaEn = Locale.CANADA;
+        if (canadaEn.getLanguage().equals(deviceLocale.getLanguage()) && canadaEn.getCountry().equals(deviceLocale.getCountry())) {
+            return "en-US";
+        }
+
+        for (String supportedTag : supportedTags) {
+            if (supportedTag.isEmpty()) continue;
+            String[] splits = supportedTag.split("-");
+            String language = splits[0];
+            if (deviceLocale.getLanguage().equals(language)) {
+                return supportedTag;
+            }
+        }
+        return "en-UK";
+    }
+
+    public String getLocaleCurrency() {
+        Locale deviceLocale;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            deviceLocale = getResources().getConfiguration().getLocales().get(0);
+            return Currency.getInstance(deviceLocale).getCurrencyCode();
+        } else {
+            deviceLocale = getResources().getConfiguration().locale;
+            return java.util.Currency.getInstance(deviceLocale).getCurrencyCode();
+        }
+    }
+
+    public boolean getLocaleMeasurementFormat() {
```

Comment:
```suggestion
    public boolean isImperialLocaleMeasurementFormat() {
```

Should probably be renamed to this (or similar) so the `boolean` return is clearer